#! /usr/bin/env python

PACKAGE='sensors'
import roslib
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#       Name                Type   Level  Description                Default  Min   Max
gen.add("enable", 			bool_t,    0,  "Enable depth completion",         	  True)
gen.add("diamondKernelSize", int_t,    0,  "Size for diamond kernel",      2,   0,   5)
gen.add("fullKernelSize",    int_t,    0,     "Size for full kernel",      1,   0,   5)
gen.add("closureKernelSize", int_t,    0,  "Size for closure kernel",      1,   0,   5)
gen.add("fillKernelSize",    int_t,    0,     "Size for fill kernel",      1,   0,   5)
gen.add("medianKernelSize",  int_t,    0,   "Size for median kernel",      2,   0,   2)

size_enum = gen.enum([ gen.const("None",      int_t, 0, "No blurring"),
                       gen.const("Gaussian",  int_t, 1, "Gaussian blurring"),
                       gen.const("Bilateral", int_t, 2, "Bilateral Filtering")],
                     "Different blurring options")
gen.add("blurMethod", int_t, 0, "Blurring method", 2, 0, 2, edit_method=size_enum)
gen.add("blurKernelSize",    int_t,    0,     "Size for blur kernel",      1,   0,   5)
gen.add("bilateralSigma", double_t,    0,"Sigma of bilateral filter",    4.5,   0,50.0)

exit(gen.generate(PACKAGE, "sensors", "DepthCompletionParams"))